log/experimental_level/level.go

```diff
@@ -1,142 +1,58 @@
-func AllowAll() []string {
-	return AllowDebugAndAbove()
-}
-
-// AllowDebugAndAbove allows all of the four default log levels.
-// Its return value may be provided with the Allowed Option.
-func AllowDebugAndAbove() []string {
-	return []string{errorLevelValue, warnLevelValue, infoLevelValue, debugLevelValue}
-}
-
-// AllowInfoAndAbove allows the default info, warn, and error log levels.
-// Its return value may be provided with the Allowed Option.
-func AllowInfoAndAbove() []string {
-	return []string{errorLevelValue, warnLevelValue, infoLevelValue}
-}
-
-// AllowWarnAndAbove allows the default warn and error log levels.
-// Its return value may be provided with the Allowed Option.
-func AllowWarnAndAbove() []string {
-	return []string{errorLevelValue, warnLevelValue}
-}
-
-// AllowErrorOnly allows only the default error log level.
-// Its return value may be provided with the Allowed Option.
-func AllowErrorOnly() []string {
-	return []string{errorLevelValue}
-}
-
-// AllowNone allows none of the default log levels.
-// Its return value may be provided with the Allowed Option.
-func AllowNone() []string {
-	return []string{}
-}
+import "github.com/go-kit/kit/log"
 
-// Error returns a logger with the level key set to ErrorLevelValue.
+// Error returns a logger that includes an error level keyval.
 func Error(logger log.Logger) log.Logger {
 	return log.NewContext(logger).WithPrefix(levelKey, errorLevelValue)
 }
 
-// Warn returns a logger with the level key set to WarnLevelValue.
+// Warn returns a logger that includes a warn level keyval.
 func Warn(logger log.Logger) log.Logger {
 	return log.NewContext(logger).WithPrefix(levelKey, warnLevelValue)
 }
 
-// Info returns a logger with the level key set to InfoLevelValue.
+// Info returns a logger that includes an info level keyval.
 func Info(logger log.Logger) log.Logger {
 	return log.NewContext(logger).WithPrefix(levelKey, infoLevelValue)
 }
 
-// Debug returns a logger with the level key set to DebugLevelValue.
+// Debug returns a logger that includes a debug level keyval.
 func Debug(logger log.Logger) log.Logger {
 	return log.NewContext(logger).WithPrefix(levelKey, debugLevelValue)
 }
 
-// New wraps the logger and implements level checking. See the commentary on the
-// Option functions for a detailed description of how to configure levels.
-// If no options are provided, all leveled log events created with level.Debug,
-// Info, Warn or Error helper methods will be squelched.
-func New(next log.Logger, options ...Option) log.Logger {
-	l := logger{
+// NewFilter wraps next and implements level filtering. See the commentary on
+// the Option functions for a detailed description of how to configure levels.
+// If no options are provided, all leveled log events created with Debug,
+// Info, Warn or Error helper methods are squelched and non-leveled log
+// events are passed to next unmodified.
+func NewFilter(next log.Logger, options ...Option) log.Logger {
+	l := &logger{
```

Comment:
I don't think there is any technical reason. The reason I moved it is because type `logger` has a pointer receiver on its `Log` methoc so it must be a pointer in the returned `log.Logger` interface value to satisfy the interface. I prefer to create it in the same form that it will live its life, as a `*logger`.